src/script/descriptor.cpp

```diff
@@ -484,6 +532,25 @@ std::unique_ptr<PubkeyProvider> ParsePubkey(const Span<const char>& sp, bool per
     return MakeUnique<BIP32PubkeyProvider>(extpubkey, std::move(path), type);
 }
 
+/** Parse a public key including origin information (if enabled). */
+std::unique_ptr<PubkeyProvider> ParsePubkey(const Span<const char>& sp, bool permit_uncompressed, FlatSigningProvider& out)
+{
+    auto colon_split = Split(sp, ':');
+    if (colon_split.size() > 2) return nullptr;
+    if (colon_split.size() == 1) return ParsePubkeyInner(colon_split[0], permit_uncompressed, out);
+    auto slash_split = Split(colon_split[0], '/');
+    if (slash_split[0].size() != 8) return nullptr;
```

Comment:
Could add CheckUnparsable checks for this condition and the !IsHex one below. Also maybe the !provider one.

Reply:
Done.